scipy/integrate/tests/test_tanhsinh.py

```diff
@@ -1096,13 +1120,17 @@ def f(k):
         b0 = a + n * step
         i = np.arange(-2, 3)
         b = b0 + i * np.spacing(b0)
-        res = nsum(f, a, b, step=step, maxterms=maxterms)
         ns = np.floor((b - a) / step)
-        assert_equal(np.diff(ns) > 0, np.diff(res.sum) > 0)
-        assert_allclose(res.sum[-1], res.sum[0] + f(b0))
         assert len(set(ns)) == 2
 
-    def test_logser_kurtosis_gh20648(self):
+        a, b = xp.asarray(a, dtype=xp.float64), xp.asarray(b, dtype=xp.float64)
+        step, ns = xp.asarray(step, dtype=xp.float64), xp.asarray(ns, dtype=xp.float64)
+        res = nsum(f, a, b, step=step, maxterms=maxterms)
+        xp_assert_equal(xp.diff(ns) > 0, xp.diff(res.sum) > 0)
```

Comment:
`diff` is not in the standard, does this matter?

Reply:
I don't think so. The function itself is incompatible with Strict because fancy indexing assignment isn't in the 2023 standard, so we skip all the tests for Strict. Besides, all the real array API libraries already have `diff`, and `diff` is in the next version of the standard.